*Cargar librerias necesarias*

In [1]:
import pandas as pd
import numpy as np

*Visualizar los datos*

In [3]:
movies = pd.read_csv('movies.csv', header="infer", sep=",")
movies.head()

MOVIES         YEAR  \
0                        Blood Red Sky       (2021)   
1  Masters of the Universe: Revelation     (2021– )   
2                     The Walking Dead  (2010–2022)   
3                       Rick and Morty     (2013– )   
4                      Army of Thieves       (2021)   

                                        GENRE  RATING  \
0      \nAction, Horror, Thriller                 6.1   
1  \nAnimation, Action, Adventure                 5.0   
2       \nDrama, Horror, Thriller                 8.2   
3  \nAnimation, Adventure, Comedy                 9.2   
4         \nAction, Crime, Horror                 NaN   

                                            ONE-LINE  \
0  \nA woman with a mysterious illness is forced ...   
1  \nThe war for Eternia begins again in what may...   
2  \nSheriff Deputy Rick Grimes wakes up from a c...   
3  \nAn animated series that follows the exploits...   
4  \nA prequel, set before the events of Army of ...   

                                               STARS    VOTES  RunTime Gross  
0  \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   NaN  
1  \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   NaN  
2  \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   NaN  
3  \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   NaN  
4  \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   NaN

*Dimensión del dataset*

In [4]:
movies.shape

(9999, 9)

*Información de la estructura del dataset*

In [5]:
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9999 entries, 0 to 9998
Data columns (total 9 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   MOVIES    9999 non-null   object 
 1   YEAR      9355 non-null   object 
 2   GENRE     9919 non-null   object 
 3   RATING    8179 non-null   float64
 4   ONE-LINE  9999 non-null   object 
 5   STARS     9999 non-null   object 
 6   VOTES     8179 non-null   object 
 7   RunTime   7041 non-null   float64
 8   Gross     460 non-null    object 
dtypes: float64(2), object(7)
memory usage: 703.2+ KB


*Verifico los datos faltantes*

In [6]:
(
movies.isna().sum().reset_index()
    .rename(columns={"index": "variable", 0: "count"})
    .assign(percent = lambda x: 100*x["count"]/movies.shape[0])
    .sort_values(by="count", ascending=True)
)  

variable  count    percent
0    MOVIES      0   0.000000
4  ONE-LINE      0   0.000000
5     STARS      0   0.000000
2     GENRE     80   0.800080
1      YEAR    644   6.440644
3    RATING   1820  18.201820
6     VOTES   1820  18.201820
7   RunTime   2958  29.582958
8     Gross   9539  95.399540

🧹 La evaluación de la completitud de los datos revela que la variable 'Gross' presenta la mayor tasa de incompletitud, con un 95.39% de datos faltantes. Dada la alta proporción de datos ausentes (cantidad total ganada a nivel mundial por las películas), se ha decidido eliminarla del análisis.

*Eliminado la columna Gross*

In [7]:
movies.drop(['Gross'],axis=1, inplace=True)

*Verificando que se haya eliminado*

In [8]:
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9999 entries, 0 to 9998
Data columns (total 8 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   MOVIES    9999 non-null   object 
 1   YEAR      9355 non-null   object 
 2   GENRE     9919 non-null   object 
 3   RATING    8179 non-null   float64
 4   ONE-LINE  9999 non-null   object 
 5   STARS     9999 non-null   object 
 6   VOTES     8179 non-null   object 
 7   RunTime   7041 non-null   float64
dtypes: float64(2), object(6)
memory usage: 625.1+ KB


*Estadísticas descriptivas*


In [9]:
movies.describe()

RATING      RunTime
count  8179.000000  7041.000000
mean      6.921176    68.688539
std       1.220232    47.258056
min       1.100000     1.000000
25%       6.200000    36.000000
50%       7.100000    60.000000
75%       7.800000    95.000000
max       9.900000   853.000000

🧹 En la variable RATING la calificación promedio es de 6.92, con una desviación estándar de 1.22, lo cual indica una variabilidad moderada alrededor de la media. La calificación mínima registrada es de 1.1, mientras que la máxima alcanza los 9.9, mostrando un rango amplio de opiniones del público.
En cuanto RUNTIME la duración promedio es de aproximadamente 68.7 minutos, con una desviación estándar de 47.26 minutos, lo que revela una gran variación en las duraciones. La duración mínima es de 1 minuto y la máxima llega a los 853 minutos, lo cual podría ser un valor atípico que representa algún contenido extenso.

*Verificar cuántos valores nulos hay en cada columna*

In [10]:
movies.isnull().sum()


MOVIES         0
YEAR         644
GENRE         80
RATING      1820
ONE-LINE       0
STARS          0
VOTES       1820
RunTime     2958
dtype: int64

*Rellenar valores nulos en la columna 'GENRE' con 'Desconocido'*

In [11]:

movies.fillna({'GENRE':'Desconocido' }, inplace=True)


*Mostrando el relleno de valores nulos*

In [12]:
movies[movies['GENRE'] == 'Desconocido']

MOVIES             YEAR  \
2124                                Rodney & Sheryl              NaN   
3144                            Our Man from Jersey              NaN   
3433                                The Upper World              NaN   
3695                                 Kod Adi: Kulüp         (2021– )   
4039                                  Oompa-Loompas              NaN   
...                                             ...              ...   
6453        Untitled Tituss Burgess/Netflix Project              NaN   
6461  One Piece: Enter Chopper at the Winter Island  (2020 TV Movie)   
8769                         Rádio Coisa Mais Linda           (2020)   
8770                         Rádio Coisa Mais Linda           (2020)   
8771                         Rádio Coisa Mais Linda           (2020)   

            GENRE  RATING                                           ONE-LINE  \
2124  Desconocido     NaN  \nBased on the unbelievable true story of seri...   
3144  Desconocido     NaN                                     \nAdd a Plot\n   
3433  Desconocido     NaN  \nEsso is caught in a deadly feud and on the v...   
3695  Desconocido     NaN                                     \nAdd a Plot\n   
4039  Desconocido     NaN  \nTV series centering on the adventures of the...   
...           ...     ...                                                ...   
6453  Desconocido     NaN                                \nPlot under wraps.   
6461  Desconocido     NaN                                     \nAdd a Plot\n   
8769  Desconocido     NaN                                     \nAdd a Plot\n   
8770  Desconocido     NaN                                     \nAdd a Plot\n   
8771  Desconocido     NaN                                     \nAdd a Plot\n   

                                                  STARS VOTES  RunTime  
2124  \n    Director:\nChloe Okuno\n| \n    Star:\nA...   NaN      NaN  
3144  \n            \n    Stars:\nMark Wahlberg, \nH...   NaN      NaN  
3433        \n            \n    Star:\nDaniel Kaluuya\n   NaN      NaN  
3695  \n            \n    Stars:\nGökçe Bahadir, \nB...   NaN      NaN  
4039                                                 \n   NaN      NaN  
...                                                 ...   ...      ...  
6453                                                 \n   NaN      NaN  
6461                                                 \n   NaN      NaN  
8769  \n    Director:\nCaito Ortiz\n| \n    Stars:\n...   NaN      NaN  
8770  \n    Director:\nCaito Ortiz\n| \n    Stars:\n...   NaN      NaN  
8771  \n    Director:\nCaito Ortiz\n| \n    Stars:\n...   NaN      NaN  

[80 rows x 8 columns]

*Eliminar espacios extra en los nombres de las columnas*

In [13]:
movies.columns = movies.columns.str.strip()


In [130]:
movies.head()

MOVIES         YEAR  \
0                        Blood Red Sky       (2021)   
1  Masters of the Universe: Revelation     (2021– )   
2                     The Walking Dead  (2010–2022)   
3                       Rick and Morty     (2013– )   
4                      Army of Thieves       (2021)   

                                        GENRE  RATING  \
0      \nAction, Horror, Thriller                 6.1   
1  \nAnimation, Action, Adventure                 5.0   
2       \nDrama, Horror, Thriller                 8.2   
3  \nAnimation, Adventure, Comedy                 9.2   
4         \nAction, Crime, Horror                 NaN   

                                            ONE-LINE  \
0  \nA woman with a mysterious illness is forced ...   
1  \nThe war for Eternia begins again in what may...   
2  \nSheriff Deputy Rick Grimes wakes up from a c...   
3  \nAn animated series that follows the exploits...   
4  \nA prequel, set before the events of Army of ...   

                                               STARS   VOTES  RunTime  
0  \n    Director:\nPeter Thorwarth\n| \n    Star...   21062    121.0  
1  \n            \n    Stars:\nChris Wood, \nSara...   17870     25.0  
2  \n            \n    Stars:\nAndrew Lincoln, \n...  885805     44.0  
3  \n            \n    Stars:\nJustin Roiland, \n...  414849     23.0  
4  \n    Director:\nMatthias Schweighöfer\n| \n  ...       0      NaN

*Reemplazar múltiples caracteres no deseados como \n, \t, y otros*

In [14]:
movies = movies.apply(lambda x: x.str.replace(r'[\n\t\r]', ' ', regex=True) if x.dtype == 'object' else x)


*Separando el director de los actores en la columna stars, se crea una columna respectivamente*

In [15]:
split_stars = movies['STARS'].str.split('|', expand=True)

movies['DIRECTOR'] = split_stars.iloc[:, :-1].apply(lambda x: ', '.join(x.dropna()), axis=1)
movies['ACTORS'] = split_stars.iloc[:, -1]


In [133]:
movies.head()

MOVIES         YEAR  \
0                        Blood Red Sky       (2021)   
1  Masters of the Universe: Revelation     (2021– )   
2                     The Walking Dead  (2010–2022)   
3                       Rick and Morty     (2013– )   
4                      Army of Thieves       (2021)   

                                       GENRE  RATING  \
0       Action, Horror, Thriller                 6.1   
1   Animation, Action, Adventure                 5.0   
2        Drama, Horror, Thriller                 8.2   
3   Animation, Adventure, Comedy                 9.2   
4          Action, Crime, Horror                 NaN   

                                            ONE-LINE  \
0   A woman with a mysterious illness is forced i...   
1   The war for Eternia begins again in what may ...   
2   Sheriff Deputy Rick Grimes wakes up from a co...   
3   An animated series that follows the exploits ...   
4   A prequel, set before the events of Army of t...   

                                               STARS   VOTES  RunTime  \
0       Director: Peter Thorwarth |      Stars: P...   21062    121.0   
1                    Stars: Chris Wood,  Sarah Mi...   17870     25.0   
2                    Stars: Andrew Lincoln,  Norm...  885805     44.0   
3                    Stars: Justin Roiland,  Chri...  414849     23.0   
4       Director: Matthias Schweighöfer |      St...       0      NaN   

                                            DIRECTOR  \
0                         Director: Peter Thorwarth    
1                    Stars: Chris Wood,  Sarah Mi...   
2                    Stars: Andrew Lincoln,  Norm...   
3                    Stars: Justin Roiland,  Chri...   
4                   Director: Matthias Schweighöfer    

                                              ACTORS  
0        Stars: Peri Baumeister,  Carl Anton Koch...  
1                                               None  
2                                               None  
3                                               None  
4        Stars: Matthias Schweighöfer,  Nathalie ...

*Eliminando columna 'STARS'*

In [16]:
movies.drop(['STARS'],axis=1, inplace=True)

In [135]:
movies.head()

MOVIES         YEAR  \
0                        Blood Red Sky       (2021)   
1  Masters of the Universe: Revelation     (2021– )   
2                     The Walking Dead  (2010–2022)   
3                       Rick and Morty     (2013– )   
4                      Army of Thieves       (2021)   

                                       GENRE  RATING  \
0       Action, Horror, Thriller                 6.1   
1   Animation, Action, Adventure                 5.0   
2        Drama, Horror, Thriller                 8.2   
3   Animation, Adventure, Comedy                 9.2   
4          Action, Crime, Horror                 NaN   

                                            ONE-LINE   VOTES  RunTime  \
0   A woman with a mysterious illness is forced i...   21062    121.0   
1   The war for Eternia begins again in what may ...   17870     25.0   
2   Sheriff Deputy Rick Grimes wakes up from a co...  885805     44.0   
3   An animated series that follows the exploits ...  414849     23.0   
4   A prequel, set before the events of Army of t...       0      NaN   

                                            DIRECTOR  \
0                         Director: Peter Thorwarth    
1                    Stars: Chris Wood,  Sarah Mi...   
2                    Stars: Andrew Lincoln,  Norm...   
3                    Stars: Justin Roiland,  Chri...   
4                   Director: Matthias Schweighöfer    

                                              ACTORS  
0        Stars: Peri Baumeister,  Carl Anton Koch...  
1                                               None  
2                                               None  
3                                               None  
4        Stars: Matthias Schweighöfer,  Nathalie ...

*Llenando espacios en blanco de 'ACTORS' y 'DIRECTORS' con Desconocido*

In [17]:
movies['ACTORS'] = movies['ACTORS'].fillna('Desconocido')
movies['DIRECTOR'] = movies['DIRECTOR'].fillna('Desconocido')


*Eliminando guiones y paréntesis*

In [18]:
movies["YEAR"] = movies["YEAR"].str.replace('(', '')
movies["YEAR"] = movies["YEAR"].str.replace(')', '')
movies["YEAR"] = movies["YEAR"].str.replace('-', '')

*Llenando los años que faltan con 0*

In [138]:
movies['YEAR'] = movies['YEAR'].fillna(0)


*Llenado 'VOTES' con 0 y 'RATING' también con 0*

In [125]:
movies['VOTES'] = movies['VOTES'].fillna('0')
movies['VOTES'] = movies['VOTES'].str.replace(',', '')
movies['VOTES'] = movies['VOTES'].astype(int) 

In [140]:
movies['RATING'] = movies['RATING'].fillna(0.0)

In [143]:
movies['RunTime'] = movies['RunTime'].fillna(0.0)

In [144]:
movies.isnull().sum()

MOVIES      0
YEAR        0
GENRE       0
RATING      0
ONE-LINE    0
VOTES       0
RunTime     0
DIRECTOR    0
ACTORS      0
dtype: int64

*Guardando el dataset limpio*

In [145]:
movies.to_csv('movies_cleaned.csv', index=False)

*última vista al dataset*

In [146]:
movies

MOVIES       YEAR  \
0                           Blood Red Sky       2021   
1     Masters of the Universe: Revelation     2021–    
2                        The Walking Dead  2010–2022   
3                          Rick and Morty     2013–    
4                         Army of Thieves       2021   
...                                   ...        ...   
9994                       The Imperfects     2021–    
9995                               Arcane     2021–    
9996                    Heart of Invictus     2022–    
9997                       The Imperfects     2021–    
9998                       The Imperfects     2021–    

                                          GENRE  RATING  \
0          Action, Horror, Thriller                 6.1   
1      Animation, Action, Adventure                 5.0   
2           Drama, Horror, Thriller                 8.2   
3      Animation, Adventure, Comedy                 9.2   
4             Action, Crime, Horror                 0.0   
...                                         ...     ...   
9994      Adventure, Drama, Fantasy                 0.0   
9995   Animation, Action, Adventure                 0.0   
9996             Documentary, Sport                 0.0   
9997      Adventure, Drama, Fantasy                 0.0   
9998      Adventure, Drama, Fantasy                 0.0   

                                               ONE-LINE   VOTES  RunTime  \
0      A woman with a mysterious illness is forced i...   21062    121.0   
1      The war for Eternia begins again in what may ...   17870     25.0   
2      Sheriff Deputy Rick Grimes wakes up from a co...  885805     44.0   
3      An animated series that follows the exploits ...  414849     23.0   
4      A prequel, set before the events of Army of t...       0      0.0   
...                                                 ...     ...      ...   
9994                                        Add a Plot        0      0.0   
9995                                        Add a Plot        0      0.0   
9996                                        Add a Plot        0      0.0   
9997                                        Add a Plot        0      0.0   
9998                                        Add a Plot        0      0.0   

                                               DIRECTOR  \
0                            Director: Peter Thorwarth    
1                       Stars: Chris Wood,  Sarah Mi...   
2                       Stars: Andrew Lincoln,  Norm...   
3                       Stars: Justin Roiland,  Chri...   
4                      Director: Matthias Schweighöfer    
...                                                 ...   
9994                    Stars: Morgan Taylor Campbel...   
9995                                                      
9996                   Director: Orlando von Einsiedel    
9997                        Director: Jovanka Vuckovic    
9998                        Director: Jovanka Vuckovic    

                                                 ACTORS  
0           Stars: Peri Baumeister,  Carl Anton Koch...  
1                                           Desconocido  
2                                           Desconocido  
3                                           Desconocido  
4           Stars: Matthias Schweighöfer,  Nathalie ...  
...                                                 ...  
9994                                        Desconocido  
9995                                        Desconocido  
9996                                Star: Prince Harry   
9997        Stars: Morgan Taylor Campbell,  Iñaki Go...  
9998        Stars: Morgan Taylor Campbell,  Jennifer...  

[9999 rows x 9 columns]